### ISSUE TITLE
Multi-architecture build of e2e and plumbing images

### ISSUE THREAD

**Issue by vdemeester** [tektoncd/plumbing#592]
In order to [support multiple architecture](https://github.com/tektoncd/pipeline/issues/856), we need to provide image in multiple
architecture for the projects's end-to-end tests. This also applies to some images built
in plumbing as we may use them on the target clusters (s390x, arm64, …) and/or during
end-to-end tests.

In an ideal world, we would use tekton to build those (aka dogfooding, …). My initial
search around this mainly ended up around `qemu`, `qemu-user-static` and `docker` (+buildkit).

See

-   [crazy-max/ghaction-docker-buildx: GitHub Action to set up Docker Buildx](https://github.com/crazy-max/ghaction-docker-buildx)
-   [hello-multiarch/image.yml at master · piwi3910/hello-multiarch](https://github.com/piwi3910/hello-multiarch/blob/master/.github/workflows/image.yml)

So the questions is how should we proceed:

-   Have a quick way of building multi-arch images using github action (as above)
-   Have a "middle" solution where we use dogfooding with some DinD, privileged containers
    and buildx
-   Find some way to replicate the github action flow in tekton (ideally without any need to
    privileged containers, but I have high doubt that it's possible)

/cc @afrittoli 
/kind feature

**Comment by barthy1** [comment]
Can first versions of the mentioned images be built using the fastest way? I guess it would be github actions. Just to unblock "other arch support" activities..
And some longer but more conceptual approach can be used after.

**Comment by vdemeester** [comment]
@barthy1 yeah, @afrittoli said the same during our productivity working group :angel:

**Comment by barthy1** [comment]
@vdemeester I'd like to suggest the first candidate for multi-arch build here is `gcr.io/tekton-nightly/github.com/tektoncd/pipeline/build-base:latest`, as it's used in [.ko.yaml ](https://github.com/tektoncd/pipeline/blob/master/.ko.yaml) for pipeline, and required to be mluti-arch to use `--platform=all` option. At this moment it is built by [gcr.io/kaniko-project/executor:v0.17.1](https://github.com/tektoncd/pipeline/blob/master/tekton/publish.yaml#L53), which doesn't support [multi-arch manifest](https://github.com/GoogleContainerTools/kaniko/issues/1102).
Do you think it makes sense to have the workflow to run Github actions build in [plumbing](https://github.com/tektoncd/plumbing) or better in [pipeline](https://github.com/tektoncd/pipeline)?

**Comment by barthy1** [comment]
One more idea, for future extensions. We can use Tekton task to start Github action - https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#workflow_dispatch :)

**Comment by chmouel** [comment]
Is there any cloud provider we could have access to ? (IBM, Google or others), where we can just spin up a VM there, inject our build script from a tekton task and get back the multi arch build ?

I understand that Github may seem easy to implement but I think we have the opportunity to show that tekton can be the right CI tool for this kind of workflow and not having to leverage another CI provider for it.
